packages/node-bridge/bridge.js

```diff
@@ -205,6 +212,25 @@ class Bridge {
     );
   }
 
+  /**
+   * @param {string} waitUntilId
+   * @param {Promise<any>} promise
+   * @param {boolean} isHandler
+   */
+  waitUntil(waitUntilId, promise, isHandler) {
```

Comment:
I wonder if this should be allowed in non-streaming mode as it may cause confusion why the response is delayed 🤔

Reply:
Yeah, I noticed that too. What are the parameters that determine which scenario happens? Streaming vs not? It wasn't clear in the code

Reply:
It's only streaming when `responseCallbackUrl` is present currently.